src/gval/utils/loading_datasets.py

```diff
@@ -421,86 +352,88 @@ def get_stac_data(
+            stack.attrs["time_aggregate"] = "min"
+        elif time_aggregate == "max":
+            stack = stack.max(dim="time")
+            stack.attrs["time_aggregate"] = "max"
         else:
-            stack = stack[0]
-            stack.attrs["time_aggregate"] = "none"
+            raise ValueError("A valid aggregate must be used for time ranges")
 
-        # Select specific bands
-        if bands is not None:
-            bands = [bands] if isinstance(bands, str) else bands
-            stack = stack.sel({"band": bands})
+        stack.rio.write_crs(crs, inplace=True)
+    else:
+        stack = stack[0]
+        stack.attrs["time_aggregate"] = "none"
 
-        band_metadata = (
-            stack.coords["raster:bands"] if "raster:bands" in stack.coords else None
-        )
-        if "band" in stack.dims:
-            og_names = [name for name in stack.coords["band"]]
-            names = [f"band_{x + 1}" for x in range(len(stack.coords["band"]))]
-            stack = stack.assign_coords({"band": names}).to_dataset(dim="band")
+    # Select specific bands
+    if bands is not None:
+        bands = [bands] if isinstance(bands, str) else bands
+        stack = stack.sel({"band": bands})
 
-            for metadata, var, og_var in zip(band_metadata, stack.data_vars, og_names):
-                _set_nodata(stack[var], metadata, nodata_fill)
-                stack[var] = _set_crs(stack[var], band_metadata)
-                stack[var].attrs["original_name"] = og_var
+    band_metadata = (
+        stack.coords["raster:bands"] if "raster:bands" in stack.coords else None
+    )
+    if "band" in stack.dims:
+        og_names = [name for name in stack.coords["band"]]
+        names = [f"band_{x + 1}" for x in range(len(stack.coords["band"]))]
+        stack = stack.assign_coords({"band": names}).to_dataset(dim="band")
 
-        else:
-            stack = stack.to_dataset(name="band_1")
-            _set_nodata(stack["band_1"], band_metadata, nodata_fill)
-            stack["band_1"] = _set_crs(stack["band_1"])
-            stack["band_1"].attrs["original_name"] = (
-                bands[0] if isinstance(bands, list) else bands
-            )
+        for metadata, var, og_var in zip(band_metadata, stack.data_vars, og_names):
+            _set_nodata(stack[var], metadata, nodata_fill)
+            stack[var] = _set_crs(stack[var], band_metadata)
+            stack[var].attrs["original_name"] = og_var
 
-        return stack
+    else:
+        stack = stack.to_dataset(name="band_1")
+        _set_nodata(stack["band_1"], band_metadata, nodata_fill)
+        stack["band_1"] = _set_crs(stack["band_1"])
+        stack["band_1"].attrs["original_name"] = (
+            bands[0] if isinstance(bands, list) else bands
+        )
 
+    return stack
 
-def _stac_to_df(stac_items: list, assets: list = None) -> pd.DataFrame:
-    """Convert a list of stac items to a DataFrame
+
+def stac_to_df(
+    stac_items: ItemCollection,
```

Comment:
Can you update variable names and docstring to the new collection input or whatever it ends up being?